src/LightSaml/SpBundle/DependencyInjection/Compiler/InjectEntityIdResolverCompilerPass.php

```
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the LightSAML SP-Bundle package.
+ *
+ * (c) Milos Tomic <tmilos@lightsaml.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace LightSaml\SpBundle\DependencyInjection\Compiler;
+
+use LightSaml\SpBundle\DependencyInjection\Compiler\Exception\InvalidEntityIdResolverException;
+use LightSaml\SpBundle\DependencyInjection\Configuration;
+use LightSaml\SpBundle\Security\Authentication\EntityId\EntityIdResolverInterface as Resolver;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+class InjectEntityIdResolverCompilerPass implements CompilerPassInterface
+{
+    /**
+     * List of service arguments where lightsaml.own.entity_id parameter is used
+     *
+     * @var array
+     */
+    private $serviceArgumentMap = [
+        'lightsaml.own.entity_descriptor_provider' => 0,
+        'lightsaml.own.credential_store.0' => 0,
+        'lightsaml.credential.credential_store' => 2,
+        'lightsaml_sp.slo.message_factory' => 0
+    ];
+
+    public function process(ContainerBuilder $container)
+    {
+        $parameterName = sprintf('%s.%s', Configuration::CONFIGURATION_NAME, Resolver::RESOLVER_NAME);
+        if (!$container->hasParameter($parameterName)) {
+            return;
+        }
+
+        $resolverName = $container->getParameter($parameterName);
+        if (!$this->validateResolver($container, $resolverName)) {
+            throw new InvalidEntityIdResolverException();
+        }
+
+        $resolverExpression = new Expression(sprintf("service('%s').resolve()", $resolverName));
+        foreach ($this->serviceArgumentMap as $serviceName => $argumentIndex) {
+            if (!$container->hasDefinition($serviceName)) {
+                continue;
```

I'm not so sure that silent pass is a good choice in this scenario. Maybe you should throw a `DefinitionNotFoundException` when definition is not present just like you throw `InvalidEntityIdResolverException` when Resolver is malformed.